openpype/lib/abstract_load_template.py

```diff
@@ -0,0 +1,305 @@
+import os
+import re
+import avalon
+from openpype.settings import get_project_settings
+from openpype.lib import Anatomy, get_linked_assets, get_loaders_by_name,\
+    collect_last_version_repres
+
+
+class AbstractTemplateLoader(object):
+    """
+    Abstraction of Template Loader.
+
+    Properties:
+        template_path : property to get current template path
+
+    Methods:
+        import_template : Abstract Method. Used to load template,
+            depending on current dcc
+        get_template_nodes : Abstract Method. Used to query nodes acting
+            as placeholders. Depending on current dcc
+    """
+
+    def __init__(self, placeholder_class):
+
+        self.import_template(self.template_path)
+
+        loaders_by_name = get_loaders_by_name()
+
+        # Skip if there is no loader
+        if not loaders_by_name:
+            self.log.warning("There are no registered loaders.")
+            return
+
+        current_asset = avalon.io.Session["AVALON_ASSET"]
+        self.populate_template(current_asset,
+                               loaders_by_name,
+                               placeholder_class)
+
+    @property
+    def template_path(self):
+        """
+        Property returning template path. Avoiding setter.
+        Getting template path from open pype settings based on current avalon
+        session and solving the path variables if needed.
+
+        Returns:
+            str: Solved template path
+
+        Raises:
+            ValueError: No profile found from settings for current avalon
+                session
+            KeyError: Could not solve path because a key does not exists
+                in avalon context
+            IOError: Solved path does not exists on current filesystem
+        """
+        project = avalon.io.Session["AVALON_PROJECT"]
+        anatomy = Anatomy(project)
+        project_settings = get_project_settings(project)
+        current_dcc = avalon.io.Session["AVALON_APP"]
```

Comment:
Please use `host_name` variable name as host is implementation of a dcc. Multiple DCCs may use same host implementation e.g. Nuke and NukeX both use `nuke` host. Hiero and Nukestudio have `hiero` host.